File: Problems/007-reverse-integer/reverse-integer.go
package problem007

import "math"

func reverse(x int) int {
	sign := 1
	if x < 0 {
		sign = -1
		x = -x
	}

	res := 0
	for x > 0 {
		res = res*10 + x%10
		x /= 10
	}

	res *= sign
	if res >= -math.MaxInt32 && res < math.MaxInt32 {
		return res
	}
	return 0
}

File: Problems/007-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        flag = 1
        if x < 0:
            flag = -1
            x = -x

        res = 0
        while x:
            res = res * 10 + x % 10
            x //= 10

        if -2**31 <= res < 2**31:
            return flag * res
        return 0
